pkg/operator/encryption/state.go

```diff
@@ -122,16 +131,35 @@ func getDesiredEncryptionState(oldEncryptionConfig *apiserverconfigv1.Encryption
 	for gr, grState := range desiredEncryptionState {
 		changed := false
 		for _, expected := range expectedReadSecrets {
-			if !hasKeyInSecret(grState.readSecrets, expected) {
+			found := false
+			for _, rs := range grState.readSecrets {
+				if equalKeyInSecret(rs, expected) {
+					found = true
+					break
+				}
+			}
+			if !found {
```

Comment:
when this can happen? both `rs` and `expected` were taken from `encryptionSecrets`

Reply:
ah, it means that a key is missing from `desiredEncryptionState`